make db-build is broken

```
2020/10/04 00:37:32 Updating python-safety-db data...
2020/10/04 00:37:32 error in python-safety-db update: failed to update python vulnerabilities: failed to decode JSON: json: cannot unmarshal object into Go value of type []python.RawAdvisory
make: *** [db-build] Error 1
Error: Makefile:77: recipe for target 'db-build' failed
Error: Process completed with exit code 2.
```

https://github.com/aquasecurity/trivy-db/actions/runs/287205703

The root case is:

* https://github.com/pyupio/safety-db/commit/b412631d3d1b8fc873804d78f75c3f187a745989 changed the format of `insecure_full.json` .
* https://github.com/RustSec/advisory-db/issues/414 changed the format of advisory-db.

--- a/pkg/vulnsrc/python/python.go
+++ b/pkg/vulnsrc/python/python.go
@@ -2,6 +2,7 @@ package python
 
 import (
 	"encoding/json"
+	"io"
 	"os"
 	"path/filepath"
 	"strings"
@@ -68,8 +69,8 @@ func (vs VulnSrc) update(repoPath string) error {
 
 	// for detecting vulnerabilities
 	var advisoryDB AdvisoryDB
-	if err = json.NewDecoder(f).Decode(&advisoryDB); err != nil {
-		return xerrors.Errorf("failed to decode JSON: %w", err)
+	if advisoryDB, err = decodeAdvisoryDB(f); err != nil {
+		return xerrors.Errorf("failed to decode AdvisoryDB: %w", err)
 	}
 
 	// for displaying vulnerability detail
@@ -135,4 +136,25 @@ func (vs VulnSrc) Get(pkgName string) ([]Advisory, error) {
 		results = append(results, advisory)
 	}
 	return results, nil
+}
+
+func decodeAdvisoryDB(r io.Reader) (AdvisoryDB, error) {
+	var obj map[string]interface{}
+	advisoryDB := AdvisoryDB{}
+	if err := json.NewDecoder(r).Decode(&obj); err != nil {
+		return nil, xerrors.Errorf("failed to decode JSON: %w", err)
+	}
+	for k, v := range obj {
+		slice, ok := v.([]interface{})
+		if !ok {
+			continue
+		}
+		b, _ := json.Marshal(slice)
+		var ad []RawAdvisory
+		if err := json.Unmarshal(b, &ad); err != nil {
+			return nil, xerrors.Errorf("failed to decode JSON: %w", err)
+		}
+		advisoryDB[k] = ad
+	}
+	return advisoryDB, nil
 }

--- a/pkg/vulnsrc/python/python.go
+++ b/pkg/vulnsrc/python/python.go
@@ -2,7 +2,6 @@ package python
 
 import (
 	"encoding/json"
-	"io"
 	"os"
 	"path/filepath"
 	"strings"
@@ -68,8 +67,8 @@ func (vs VulnSrc) update(repoPath string) error {
 	defer f.Close()
 
 	// for detecting vulnerabilities
-	var advisoryDB AdvisoryDB
-	if advisoryDB, err = decodeAdvisoryDB(f); err != nil {
+	advisoryDB := AdvisoryDB{}
+	if err = json.NewDecoder(f).Decode(&advisoryDB); err != nil {
 		return xerrors.Errorf("failed to decode AdvisoryDB: %w", err)
 	}
 
@@ -138,23 +137,20 @@ func (vs VulnSrc) Get(pkgName string) ([]Advisory, error) {
 	return results, nil
 }
 
-func decodeAdvisoryDB(r io.Reader) (AdvisoryDB, error) {
-	var obj map[string]interface{}
-	advisoryDB := AdvisoryDB{}
-	if err := json.NewDecoder(r).Decode(&obj); err != nil {
-		return nil, xerrors.Errorf("failed to decode JSON: %w", err)
+func (ad AdvisoryDB) UnmarshalJSON(data []byte) error {
+	var obj map[string]json.RawMessage
+	if err := json.Unmarshal(data, &obj); err != nil {
+		return xerrors.Errorf("failed to decode JSON: %w", err)
 	}
 	for k, v := range obj {
-		slice, ok := v.([]interface{})
-		if !ok {
+		if k == "$meta" {
 			continue
 		}
-		b, _ := json.Marshal(slice)
-		var ad []RawAdvisory
-		if err := json.Unmarshal(b, &ad); err != nil {
-			return nil, xerrors.Errorf("failed to decode JSON: %w", err)
+		var raw []RawAdvisory
+		if err := json.Unmarshal(v, &raw); err != nil {
+			return xerrors.Errorf("failed to decode JSON: %w", err)
 		}
-		advisoryDB[k] = ad
+		ad[k] = raw
 	}
-	return advisoryDB, nil
+	return nil
 }

--- a/pkg/vulnsrc/python/python.go
+++ b/pkg/vulnsrc/python/python.go
@@ -69,7 +69,7 @@ func (vs VulnSrc) update(repoPath string) error {
 	// for detecting vulnerabilities
 	advisoryDB := AdvisoryDB{}
 	if err = json.NewDecoder(f).Decode(&advisoryDB); err != nil {
-		return xerrors.Errorf("failed to decode AdvisoryDB: %w", err)
+		return xerrors.Errorf("failed to decode JSON: %w", err)
 	}
 
 	// for displaying vulnerability detail

--- a/pkg/vulnsrc/python/python_test.go
+++ b/pkg/vulnsrc/python/python_test.go
@@ -95,7 +95,8 @@ func TestVulnSrc_Update(t *testing.T) {
 }
 
 func Test_decodeAdvisoryDB(t *testing.T) {
-	b, _ := ioutil.ReadFile("testdata/fixtures/python-safety-db/data/insecure_full.json")
+	b, err := ioutil.ReadFile("testdata/fixtures/python-safety-db/data/insecure_full.json")
+	require.NoError(t, err)
 	advisories := AdvisoryDB{}
 	err := json.Unmarshal(b, &advisories)
 	require.NoError(t, err)

--- a/pkg/vulnsrc/python/python_test.go
+++ b/pkg/vulnsrc/python/python_test.go
@@ -140,5 +140,5 @@ func Test_decodeAdvisoryDB(t *testing.T) {
 			},
 		},
 	}
-	require.Equal(t, expectedAdvisory, advisories, "expected %v, got: %v", expectedAdvisory, advisories)
+	assert.Equal(t, expectedAdvisory, advisories)
 }

--- a/pkg/vulnsrc/python/python_test.go
+++ b/pkg/vulnsrc/python/python_test.go
@@ -1,7 +1,6 @@
 package python
 
 import (
-	"bytes"
 	"encoding/json"
 	"io/ioutil"
 	"os"
@@ -97,8 +96,8 @@ func TestVulnSrc_Update(t *testing.T) {
 
 func Test_decodeAdvisoryDB(t *testing.T) {
 	b, _ := ioutil.ReadFile("testdata/fixtures/python-safety-db/data/insecure_full.json")
-	r := bytes.NewReader(b)
-	advisories, err := decodeAdvisoryDB(r)
+	advisories := AdvisoryDB{}
+	err := json.Unmarshal(b, &advisories)
 	require.NoError(t, err)
 	expectedAdvisory := AdvisoryDB{
 		"zulip": []RawAdvisory{

--- a/pkg/vulnsrc/python/python_test.go
+++ b/pkg/vulnsrc/python/python_test.go
@@ -1,7 +1,9 @@
 package python
 
 import (
+	"bytes"
 	"encoding/json"
+	"io/ioutil"
 	"os"
 	"sort"
 	"strings"
@@ -91,4 +93,53 @@ func TestVulnSrc_Update(t *testing.T) {
 			require.Equal(t, tc.expectedAdvisory, advisories, "expected %v for package zulip, got: %v", tc.expectedAdvisory, advisories)
 		})
 	}
+}
+
+func Test_decodeAdvisoryDB(t *testing.T) {
+	b, _ := ioutil.ReadFile("testdata/fixtures/python-safety-db/data/insecure_full.json")
+	r := bytes.NewReader(b)
+	advisories, err := decodeAdvisoryDB(r)
+	require.NoError(t, err)
+	expectedAdvisory := AdvisoryDB{
+		"zulip": []RawAdvisory{
+			{
+				ID:       "pyup.io-38116",
+				Advisory: "Zulip 2.0.7 inlcudes a fix for insecure account creation via social authentication - see CVE-2019-18933. It also adds backend enforcement of zxcvbn password strength checks.",
+				Cve:      "CVE-2019-18933",
+				Specs:    []string{"<2.0.7"},
+				Version:  "<2.0.7",
+			},
+			{
+				ID:       "pyup.io-38200",
+				Advisory: "Zulip 2.1.3 includes fixes for:\r\n- CVE-2020-9444: Prevent reverse tabnapping attacks.                                                 \r\n- CVE-2020-9445: Remove unused and insecure modal_link feature.                                      \r\n- CVE-2020-10935: Fix XSS vulnerability in local link rewriting.",
+				Cve:      "CVE-2020-9444,CVE-2020-10935",
+				Specs:    []string{"<2.1.3"},
+				Version:  "<2.1.3",
+			},
+			{
+				ID:       "pyup.io-36168",
+				Advisory: "Zulip 1.7.2 is a security release, with a handful of cherry-picked changes since 1.7.1.\r\n- CVE-2018-9986: Fix XSS issues with frontend markdown processor.\r\n- CVE-2018-9987: Fix XSS issue with muting notifications.\r\n- CVE-2018-9990: Fix XSS issue with stream names in topic typeahead.\r\n- CVE-2018-9999: Fix XSS issue with user uploads.  The fix for this adds a Content-Security-Policy for the `LOCAL_UPLOADS_DIR` storage backend for user-uploaded files.",
+				Cve:      "CVE-2018-9986,CVE-2018-9987,CVE-2018-9990,CVE-2018-9999",
+				Specs:    []string{"<1.7.2"},
+				Version:  "<1.7.2",
+			},
+		},
+		"kuber": []RawAdvisory{
+			{
+				ID:       "pyup.io-38099",
+				Advisory: "Kuber 10.0.1 bumps the urllib3 version to pick up security fix for CVE-2019-11324.",
+				Cve:      "CVE-2019-11324",
+				Specs:    []string{"<10.0.1"},
+				Version:  "<10.0.1",
+			},
+			{
+				ID:       "pyup.io-36979",
+				Advisory: "kuber 9.0.0a1 bumps urllib3 version to pick up security fix for CVE-2018-20060.",
+				Cve:      "",
+				Specs:    []string{"<9.0.0a1"},
+				Version:  "<9.0.0a1",
+			},
+		},
+	}
+	require.Equal(t, expectedAdvisory, advisories, "expected %v, got: %v", expectedAdvisory, advisories)
 }

--- a/pkg/vulnsrc/python/python_test.go
+++ b/pkg/vulnsrc/python/python_test.go
@@ -94,7 +94,7 @@ func TestVulnSrc_Update(t *testing.T) {
 	}
 }
 
-func Test_decodeAdvisoryDB(t *testing.T) {
+func TestAdvisoryDB_ UnmarshalJSON(t *testing.T) {
 	b, err := ioutil.ReadFile("testdata/fixtures/python-safety-db/data/insecure_full.json")
 	require.NoError(t, err)
 	advisories := AdvisoryDB{}

--- a/pkg/vulnsrc/python/python_test.go
+++ b/pkg/vulnsrc/python/python_test.go
@@ -9,6 +9,7 @@ import (
 	"testing"
 
 	"github.com/aquasecurity/trivy-db/pkg/db"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
@@ -94,11 +95,11 @@ func TestVulnSrc_Update(t *testing.T) {
 	}
 }
 
-func TestAdvisoryDB_ UnmarshalJSON(t *testing.T) {
+func TestAdvisoryDB_UnmarshalJSON(t *testing.T) {
 	b, err := ioutil.ReadFile("testdata/fixtures/python-safety-db/data/insecure_full.json")
 	require.NoError(t, err)
 	advisories := AdvisoryDB{}
-	err := json.Unmarshal(b, &advisories)
+	err = json.Unmarshal(b, &advisories)
 	require.NoError(t, err)
 	expectedAdvisory := AdvisoryDB{
 		"zulip": []RawAdvisory{

--- a/pkg/vulnsrc/python/testdata/fixtures/python-safety-db/data/insecure_full.json
+++ b/pkg/vulnsrc/python/testdata/fixtures/python-safety-db/data/insecure_full.json
@@ -1,4 +1,8 @@
 {
+  "$meta": {
+    "advisory": "PyUp.io metadata",
+    "timestamp": 1601532000
+  },
   "zulip": [
     {
       "advisory": "Zulip 2.0.7 inlcudes a fix for insecure account creation via social authentication - see CVE-2019-18933. It also adds backend enforcement of zxcvbn password strength checks.",
@@ -26,6 +30,26 @@
         "<1.7.2"
       ],
       "v": "<1.7.2"
+    }
+  ],
+  "kuber": [
+    {
+      "advisory": "Kuber 10.0.1 bumps the urllib3 version to pick up security fix for CVE-2019-11324.",
+      "cve": "CVE-2019-11324",
+      "id": "pyup.io-38099",
+      "specs": [
+        "<10.0.1"
+      ],
+      "v": "<10.0.1"
+    },
+    {
+      "advisory": "kuber 9.0.0a1 bumps urllib3 version to pick up security fix for CVE-2018-20060.",
+      "cve": null,
+      "id": "pyup.io-36979",
+      "specs": [
+        "<9.0.0a1"
+      ],
+      "v": "<9.0.0a1"
     }
   ]
 }

